src/util/path.js

```diff
@@ -590,6 +590,13 @@
         currentPath,
         parsed,
         re = fabric.rePathCommand,
+        rCommaWsp = '(?:\\s,?\\s*|,\\s*)',
+        rNumberCommaWsp = '(' + fabric.reNum + ')' + rCommaWsp,
+        rFlagCommaWsp = '([01])' + rCommaWsp + '?',
+        rCoordinatePair = '(' + fabric.reNum + ')' + rCommaWsp + '?(' + fabric.reNum + ')',
+        rArcSeq = rNumberCommaWsp + '?' + rNumberCommaWsp + '?' + rNumberCommaWsp + rFlagCommaWsp + rFlagCommaWsp +
+          rCoordinatePair,
```

Comment:
isn't rCoordinatePair just another 2 `rNumberCommaWsp`?

is `rArcSeq` just:

```js
rArcSeq = rNumberCommaWsp + '?' + rNumberCommaWsp + '?' + rNumberCommaWsp + rFlagCommaWsp + rNumberCommaWsp + '?'  + rNumberCommaWsp;
```
or am i over simplifying?

Also, what the `?` is doing? i see is between everything but not after the third `rNumberCommaWsp`

Reply:
I'll change up the code. rCoordinatePair is rNumberCommaWsp + '?' + rNumber (no comma/spaces after). We don't need the abstraction unless we want to be clear we're matching the grammar with variable names.

The grammar claims the first two have optional commaspaces, which is what that '?' is handling (we'd need a different variable if '?' handled the whole rNumberCommaWsp but we don't want that). I question how accurate that is but matching the grammar is probably best here.